src/drivers/audio/stm32/stm32_audio_init.c

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file
+ * @brief Driver for STM32
+ * @author Alex Kalmuk
+ * @date 07.11.2018
+ */
+
+#include <embox/unit.h>
+#include <util/log.h>
+
+#include <drivers/audio/stm32_audio.h>
+
+EMBOX_UNIT_INIT(stm32_audio_init);
+
+static int stm32_rate = 0;
+
+/* XXX
+ * STM32 Cube suppose only 3 possible audio configurations:
+ * 1. BSP_AUDIO_OUT_Play - only OUTPUT
+ * 2. BSP_AUDIO_IN_Record - only INPUT
+ * 3. BSP_AUDIO_IN_OUT_Init - BOTH
+ * So the only one possibility to use both audio in and audio out is
+ * to use BSP_AUDIO_IN_OUT_Init, which makes run BOTH at any time. It means
+ * that even we use 'record' audio out will be activated too.
+ * WORKAROUND: Run both in adn out everytime, and actually do not stop them
+ * even if auddio_dev->stop() is called.
+ */
+
+static int stm32_audio_init(void) {
+	if (0 != irq_attach(STM32_AUDIO_IN_DMA_IRQ, stm32_audio_in_dma_interrupt,
+				0, NULL, "stm32_audio_dma_in")) {
+		log_error("irq_attach error");
+	}
+	if (0 != irq_attach(STM32_AUDIO_OUT_IRQ, stm32_audio_i2s_dma_interrupt,
+				0, NULL, "stm32_audio_out")) {
+		log_error("irq_attach error");
+	}
+
+	/* Set volume 90 for both audio input and output. Then we can change
+	 * output volume in stm32_audio_out_start() */
+	if (0 != BSP_AUDIO_IN_OUT_Init(
+				INPUT_DEVICE_DIGITAL_MICROPHONE_2,
+				OUTPUT_DEVICE_HEADPHONE,
+				90, 16000)) {
+		log_error("BSP_AUDIO_IN_OUT_Init error");
+		return -1;
+	}
+
+	stm32_audio_out_init_once();
+	stm32_audio_in_init_once();
+
+	return 0;
+}
+
+int stm32_audio_get_rate(void) {
+	return stm32_rate;
+}
+
+void stm32_audio_set_rate(int rate) {
```

Comment:
This function doesn't change actual rate of input.

Is it ok?

Also, it has single variable `stm32_rate` for both input and output.